connectors/sources/generic_database.py

```diff
@@ -0,0 +1,444 @@
+            query (str): Database query to be executed.
+            engine (Engine): Engine to connect with the Database server
+
+        Returns:
+            cursor: Synchronous cursor
+        """
+        try:
+            loop = asyncio.get_running_loop()
+            self.connection = await loop.run_in_executor(
+                executor=None, func=engine.connect
+            )
+            cursor = await loop.run_in_executor(
+                executor=None, func=partial(self.connection.execute, statement=query)
+            )
+            return cursor
+        except Exception as exception:
+            logger.warning(
+                f"Something went wrong while executing query. Exception: {exception}"
+            )
+            raise
+
+    def serialize(self, doc):
+        """Reads each element from the document and serialize it as per it's datatype.
+
+        Args:
+            doc (Dict): Dictionary to be serialize
+
+        Returns:
+            doc (Dict): Serialized version of dictionary
+        """
+
+        def _serialize(value):
+            """Serialize input value as per it's datatype.
+            Args:
+                value (Any Datatype): Value to be serialize
+
+            Returns:
+                value (Any Datatype): Serialized version of input value.
+            """
+
+            if isinstance(value, (list, tuple)):
+                value = [_serialize(item) for item in value]
+            elif isinstance(value, dict):
+                for key, svalue in value.items():
+                    value[key] = _serialize(svalue)
+            elif isinstance(value, (datetime, date)):
+                value = value.isoformat()
+            elif isinstance(value, Decimal128):
+                value = value.to_decimal()
+            elif isinstance(value, (bytes, bytearray)):
+                value = value.decode(errors="ignore")
+            elif isinstance(value, Decimal):
+                value = float(value)
+            return value
+
+        for key, value in doc.items():
+            doc[key] = _serialize(value)
+
+        return doc
+
+    async def fetch_documents(self, table, engine, is_async=True, schema=None):
+        """Fetches all the table entries and format them in Elasticsearch documents
+
+        Args:
+            table (str): Name of table
+            engine (Engine): Engine to connect with the Database server
+            is_async (bool, optional): Flag to indicates async/sync execution. Defaults to True.
+            schema (str): Name of schema. Defaults to None.
+
+        Yields:
+            Dict: Document to be indexed
+        """
+        try:
+            [[row_count]] = await anext(
+                self.execute_query(
+                    query_name="TABLE_DATA_COUNT",
+                    engine=engine,
+                    is_async=is_async,
+                    schema=schema,
+                    table=table,
+                )
+            )
+            if row_count:
```

Comment:
`row_count` is not a boolean and a rather complex structure.. (`[[]]`) can we change this for an explicit test?

Reply:
Done